src/vec_int128_ppc.h

```diff
@@ -2022,6 +2310,146 @@ vec_divsq_10e31 (vi128_t vra)
+ *  the constant 10e31. The final remainder can be passed to
+ *  <B>Decimal Convert From Signed Quadword</B> (See vec_bcdcfsq()).
+ *  Long division it repeated on the resulting multi-quadword quotient
+ *  to extract 31-digits for each step. This is repeated until the
+ *  multi-quadword quotient to result to less than 10e31 which
+ *  provides the highest order 31-digits of the of the multiple
+ *  precision binary to BCD conversion.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   | 12-192| 1/cycle  |
+ *  |power9   | 9-127 | 1/cycle  |
+ *
+ *  @param *qh the high quotient as a vector unsigned __int128.
+ *  @param vra the high dividend as a vector unsigned __int128.
+ *  @param vrb the low dividend as a vector unsigned __int128.
+ *  @return the low quotient as vector unsigned __int128.
+ */
+static inline vui128_t
+vec_divudq_10e31 (vui128_t *qh, vui128_t vra, vui128_t vrb)
+{
+  const vui128_t ten31 = (vui128_t)
+	  { (__int128) 1000000000000000UL * (__int128) 10000000000000000UL };
+  const vui128_t zero = (vui128_t) { (__int128) 0UL };
+  /* Magic numbers for multiplicative inverse to divide by 10**31
+   are 4804950418589725908363185682083061167, corrective add,
+   and shift right 103 bits.  */
+  const vui128_t mul_invs_ten31 = (vui128_t) CONST_VINT128_DW(
+      0x039d66589687f9e9UL, 0x01d59f290ee19dafUL);
+  const int shift_ten31 = 103;
+  vui128_t result, r2, t, q, q1, q2, c;
+
+  if (vec_cmpuq_all_ne (vra, zero) || vec_cmpuq_all_ge (vrb, ten31))
+    {
+      // Multiply high [vra||vrb] * mul_invs_ten31
+      q = vec_mulhuq (vrb, mul_invs_ten31);
+      q1 = vec_muludq (&t, vra, mul_invs_ten31);
+      c = vec_addcuq (q1, q);
+      q = vec_adduqm (q1, q);
+      q1 = vec_adduqm (t, c);
+      // corrective add [q2||q1||q] = [q1||q] + [vra||vrb]
+      c = vec_addcuq (vrb, q);
+      q = vec_adduqm (vrb, q);
+      // q2 is the carry-out from the corrective add
+      q2 = vec_addecuq (q1, vra, c);
+      q1 = vec_addeuqm (q1, vra, c);
+      // shift 384-bits (including the carry) right 107 bits
+      // Using shift left double quadword shift by (128-107)-bits
+      r2 = vec_sldqi (q2, q1, (128 - shift_ten31));
+      result = vec_sldqi (q1, q, (128 - shift_ten31));
+    }
+  else
+    {
+      // Dividend less than divisor then return zero quotient
+      r2 = zero;
+      result = zero;
+    }
+
+  // return 256-bit quotient
+  *qh = r2;
+  return result;
+}
+
+
+/** \brief Vector Divide Unsigned Double Quadword by const 10e32.
+ *
+ *  Compute the quotient of 256 bit value vra||vrb / 10e32.
+ *
+ *  \note vec_divudq_10e32() and vec_modudq_10e32() can be used
+ *  to perform long division of a multi-quaqword binary value by
+ *  the constant 10e32. The final remainder can be passed to
+ *  <B>Decimal Convert From Unsigned Quadword</B> (See vec_bcdcfuq()).
+ *  Long division it repeated on the resulting multi-quadword quotient
```

Comment:
copy and paste from above... s/it/is/, "repeated" is repeated, and s/to less/is less/

Reply:
yes